packages/ember-simple-auth-cookie-store/lib/simple-auth-cookie-store/stores/cookie.js

```diff
@@ -181,5 +208,58 @@ export default Base.extend({
       Ember.run.cancel(this._syncDataTimeout);
       this._syncDataTimeout = Ember.run.later(this, this.syncData, 500);
     }
+  },
+
+  /**
+    @method pageVisibilityInit
+    @private
+  */
+  pageVisibilityInit: function(){
+    var stateKey;
+    var eventKey;
+    var keys = {
+      hidden: "visibilitychange",
+      webkitHidden: "webkitvisibilitychange",
+      mozHidden: "mozvisibilitychange",
+      msHidden: "msvisibilitychange"
+    };
+    for (stateKey in keys) {
+      if (stateKey in document) {
+        eventKey = keys[stateKey];
+        break;
+      }
+    }
+    return function(c) {
+      if (c) {document.addEventListener(eventKey, c);}
+      return !document[stateKey];
+    };
+  },
```

Comment:
I guess I'm just too stupid but can you explain me how this works? How does it detect which one of `'visibilitychange'`, `'webkitvisibilitychange'` etc. to use? Also how does it detect when the visibility changes?